openapi-cli/src/main/java/io/ballerina/openapi/generators/schema/BallerinaSchemaGenerator.java

```diff
@@ -636,15 +637,15 @@ public TypeDescriptorNode getTypeDescriptorNodeForArraySchema(OpenAPI openApi, A
         Token openSBracketToken = AbstractNodeFactory.createIdentifierToken("[");
         Token closeSBracketToken = AbstractNodeFactory.createIdentifierToken("]");
         Schema<?> schemaItem = arraySchema.getItems();
-        if (schemaItem.get$ref() != null) {
-            type = getValidName(extractReferenceType(arraySchema.getItems().get$ref()), true);
-            if (arraySchema.getNullable() != null) {
-                if (arraySchema.getNullable()) {
-                    closeSBracketToken = AbstractNodeFactory.createIdentifierToken("]?");
-                }
-            } else if (nullable) {
+        if (arraySchema.getNullable() != null) {
+            if (arraySchema.getNullable()) {
                 closeSBracketToken = AbstractNodeFactory.createIdentifierToken("]?");
             }
```

Comment:
Here we have the explicit need of ignoring the `nullable: false` scenario. If we added the condition suggested in the `arraySchema.getNullable() != null` and `arraySchema.getNullable() == false ` and `nullable: true` scenario, the particular field will bet set to `nillable`. But since it has been specifically mentioned as `nullable: false` in the OpenAPI this outcome would be a spec violation.

Reply:
Ack, the previous way make sense for the avoid spec violation. :)